docker-images/oncoanalyser/assets/run.sh

```diff
@@ -165,79 +177,175 @@ fi
 missing_args=()
 for argname in ${required_args}; do
   if [[ -z "${!argname:-}" ]]; then
-    missing_args+=(${argname})
+    missing_args+=( "${argname}" )
   fi
 done
 
 if [[ ${#missing_args[@]} -gt 0 ]]; then
   print_help_text
-  missing_arg_str=$(echo -n "--${missing_args[0]}"; for arg in ${missing_args[@]:1}; do echo -n ", --${arg}"; done)
-  plurality=$(if [[ ${#missing_args[@]} -gt 1 ]]; then echo are; else echo is; fi)
+
+  # Get missing arguments
+  missing_arg_str="$( \
+    echo -n "--${missing_args[0]}";
+    for arg in "${missing_args[@]:1}"; do
+      echo -n ", --${arg}";
+    done \
+  )"
+
+  # More than one missing arg?
+  # Lets make sure our grammar is right
+  plurality="$( \
+    if [[ "${#missing_args[@]}" -gt 1 ]]; then
+      echo are;
+      else echo is;
+    fi \
+  )"
   echo -e "\nERROR: ${missing_arg_str} ${plurality} required" 1>&2
   exit 1
 fi
 
-# When Nextflow runs a job using the local executor with Docker enabled, I have configured behaviour such that that a
-# new Docker container from the host service is launched. This means that all local Nextflow processes inherit the EC2
-# instance IAM profile, which can only be set prior at the Batch compute environment creation; in this case that is the
-# non-permissive Nextflow pipeline role. This means to run Nextflow processes locally that can r/w to S3 (e.g. when
-# using Fusion, S3 output directory, etc), we must set the EC2 instance IAM role to a profile with such permissions.
-# Here I associate the instance with the OncoanalyserStack task role. There may be better approaches to achieve this.
+## END USAGE / ARGS ##
 
-# NOTE(SW): this is to be updated manually for now until build process is placed into CodePipeline
+## FUNCTIONS ##
 
-instance_id=$(curl -s http://169.254.169.254/latest/meta-data/instance-id)
-association_id=$(
-  aws ec2 describe-iam-instance-profile-associations | \
-    jq -r '
-      .IamInstanceProfileAssociations[] |
-        select(.InstanceId == "'${instance_id}'") |
-        .AssociationId
+## SSM Parameter functions
+get_ssm_parameter_value(){
+  aws ssm get-parameter \
+    --name "$1" \
+    --output json |
+  jq '.Parameter | .Value'
+}
+
+get_cache_bucket_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/nxf/cache_bucket"
+}
+
+get_cache_path_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/nxf/cache_path"
+}
+
+get_dest_bucket_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/nxf/staging_bucket"
+}
+
+get_dest_prefix_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/nxf/staging_prefix"
+}
+
+get_hmf_refdata_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/reference_data/hmf"
+}
+
+get_hmf_virusbreakend_db_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/reference_data/virusbreakend"
+}
+
+get_hmf_genomes_path_from_ssm(){
+  get_ssm_parameter_value "/oncoanalyser/reference_data/genomes"
+}
```

Comment:
Also dropping `hmf` for both the VIRUSBreakend database and reference genomes function name would be helpful here.

Reply:
Resolved in 7acc9c13da2d7887fd0065cbb5f7971d1f22d4dc